modules/mono/godotsharp_dirs.cpp

```diff
@@ -139,19 +135,52 @@ class _GodotSharpDirs {
 		String arch = Engine::get_singleton()->get_architecture_name();
 		String appname = GLOBAL_GET("application/config/name");
 		String appname_safe = OS::get_singleton()->get_safe_dir_name(appname);
-		String data_dir_root = exe_dir.path_join("data_" + appname_safe + "_" + arch);
-		if (!DirAccess::exists(data_dir_root)) {
-			data_dir_root = exe_dir.path_join("data_Godot_" + arch);
-		}
+		String packed_path = "res://.godot/mono/publish/" + arch;
+		if (DirAccess::exists(packed_path)) {
+			// The dotnet publish data is packed in the pck/zip.
```

Comment:
Is this only relevant for Android?

Reply:
As this PR is atm yes, but mostly because I didn't want to take a dependency on the export options PR. With that putting the data files into the pck would be relevant for all platforms.